nexus/db-queries/src/db/datastore/instance.rs

```diff
@@ -373,19 +373,32 @@ impl DataStore {
 
         use db::schema::instance::dsl as instance_dsl;
         use db::schema::migration::dsl as migration_dsl;
-        use db::schema::vmm::dsl as vmm_dsl;
+        use db::schema::vmm;
+
+        // Create a Diesel alias to allow us to LEFT JOIN the `instance` table
+        // with the `vmm` table twice; once on the `active_propolis_id` and once
+        // on the `target_propolis_id`.
+        let (active_vmm, target_vmm) =
+            diesel::alias!(vmm as active_vmm, vmm as target_vmm);
+        let vmm_selection =
+            <Vmm as Selectable<diesel::pg::Pg>>::construct_selection();
```

Comment:
Great find! I looked for something like this hanging off of `Alias` and came up empty.

Reply:
Yeah, there was, as far as I can tell, no actual documentation explaining this usage  I found it by looking at the implementation of `as_select()` and noticed that there was a comment on the `Selectable` trait saying that the return value from `construct_selection()` was "usually a tuple of fields".